tests/raftstore/test_split_region.rs

```diff
@@ -434,3 +436,52 @@ fn test_server_split_with_stale_peer() {
     let mut cluster = new_server_cluster(0, 3);
     test_split_with_stale_peer(&mut cluster);
 }
+
+fn test_split_region_diff_check<T: Simulator>(cluster: &mut Cluster<T>) {
+    let region_max_size = 2000;
+    let region_split_size = 1000;
+    cluster.cfg.store_cfg.split_region_check_tick_interval = 100;
+    cluster.cfg.store_cfg.region_check_size_diff = 10;
+    cluster.cfg.store_cfg.region_max_size = region_max_size;
+    cluster.cfg.store_cfg.region_split_size = region_split_size;
+    cluster.cfg.store_cfg.raft_log_gc_tick_interval = 20000;
+
+    let mut range = 1..;
+
+    cluster.run();
+
+    let pd_client = cluster.pd_client.clone();
+
+    put_till_size(cluster, region_max_size * 10, &mut range);
+
+    let min_region_cnt = (region_max_size * 10 - region_max_size) / region_split_size + 2;
```

Comment:
Peer will split when size of region meet region_max_size, so assume the last region_max_size of data is not involved in split, there will be at least `(region_max_size * 10 - region_max_size) / region_split_size` regions. But region_max_size of data should be split too, so there will be at least 2 more regions. So the region count should not be less than `(region_max_size * 10 - region_max_size) / region_split_size + 2`.